app/main.py

```diff
@@ -1 +1,45 @@
-#  write your code here
+from typing import List
+
+
+class Animal:
+    def __init__(self,
+                 name: str,
+                 appetite: int,
+                 is_hungry: bool = True
+                 ) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> str:
+        print("Hello, I'm", self.name)
+
+    def feed(self) -> str:
+        if self.is_hungry:
+            print("Eating", self.appetite, "food points...")
+            self.is_hungry = False
+            return self.appetite
+        return 0
+
+
+class Cat(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, 3, is_hungry)
+
+    def catch_mouse(self) -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, appetite=7, is_hungry=is_hungry)
```

Comment:
Stick to one approach, either passing `args` or `kwargs` here in the `__init__`. It should align with the `__init__` from line 27 as well.

Reply:
Done